module/zfs/zio.c

```diff
@@ -1706,6 +1706,15 @@ zio_write_compress(zio_t *zio)
 	if (compress != ZIO_COMPRESS_OFF &&
 	    !(zio->io_flags & ZIO_FLAG_RAW_COMPRESS)) {
 		void *cbuf = zio_buf_alloc(lsize);
+		/*
+		 * Both the crypto code and LZ4 code have bad habits of reading
+		 * uninited bytes sometimes.
+		 *
+		 * So to stop valgrind screaming to the point of uselessness,
+		 * zero them all.
+		 */
+		bzero(cbuf, lsize);
```

Comment:
As a preventative measure for rebase at a later date (#12996): `memset(cbuf, 0, lsize);`

Reply:
Done!